src/array/ops/filter.rs

```diff
@@ -46,6 +50,13 @@ impl NullArray {
     }
 }
 
+impl FixedSizeListArray {
+    pub fn filter(&self, mask: &BooleanArray) -> DaftResult<Self> {
```

Comment:
I think so! I think that making all of these array function traits, in case we want to use them internally in other kernels, since you can then define the other kernel as generic over that trait.